### PR TITLE
Hyundai: Add FW Versions for 2019 Elantra

### PR DESCRIPTION
**Pull request by sunnyhaibin**
Add firmware versions for the 2019 Hyundai Elantra.

Dongle ID: `8d74700a3ad7f97e`

Thanks to the community 2019 Hyundai Elantra owner kingdingdong(Discord).

### CODE REVIEW

File: selfdrive/car/hyundai/values.py [hunk 1]
```diff
@@ -1619,6 +1619,7 @@ def match_fw_to_car_fuzzy(live_fw_versions) -> Set[str]:
     ],
     (Ecu.fwdCamera, 0x7c4, None): [
       b'\xf1\x00JFP LKAS AT EUR LHD 1.00 1.03 95895-A8100 160711',
+      b'\xf1\x00ADP LKAS AT USA LHD 1.00 1.03 99211-F2000 X31',
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:1622]: Why is this on the OptimaH?

**Comment by sshane** [comment]
Nice, finally adding platform codes for the Elantra. I think the current platform code we have corresponds to the Hyundai i30.

File: selfdrive/car/hyundai/values.py (hunk 1 above)
Comment (sunnyhaibin) [reply selfdrive/car/hyundai/values.py]: Wrong platform, fixed :)

**Comment by sunnyhaibin** [comment]
The user mentioned that Harness G was the one that he got working with.

**Comment by sshane** [comment]
Looks like 2017, 2018 were best guesses, and then expanded to 2019. Fine to change.

File: selfdrive/car/hyundai/values.py [hunk 2]
```diff
@@ -1637,11 +1639,16 @@ def match_fw_to_car_fuzzy(live_fw_versions) -> Set[str]:
...
+      b'\xf1\x00AD ESC \x11 11 \x18\x05\x06 58910-F2840',
     ],
     (Ecu.fwdRadar, 0x7d0, None): [
       b'\xf1\x00PD__ SCC F-CUP      1.00 1.00 96400-G3300         ',
       b'\xf1\x00PD__ SCC FNCUP      1.01 1.00 96400-G3000         ',
+      b'\xf1\x00AD__ SCC H-CUP      1.00 1.00 99110-F2100         ',
     ],
+    (Ecu.engine, 0x7e0, None): [
+      b'\xf1\x8161698051\x00\x00\x00\x00\x00\x00\x00\x00',
+    ]
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:1651]: You can't add a new ECU since now the older MY cars will be missing this and not fingerprint.

Comment (sshane) [reply selfdrive/car/hyundai/values.py:1651]: The other Elantra is an Elantra GT 2019 with no engine, so it seems split not only on MY but also model. Going to remove for now.

Comment (sshane) [inline selfdrive/car/hyundai/values.py:1651]:
```suggestion
```

**Comment by sshane** [comment]
This car also doesn't return EPS (needs different query like Optima), and transmission response is spotty (it changed to `b'\xf1\x006T6K0_C2\x00\x006T6S2051\x00\x00TAD0N20NSD\x00\x00\x00\x00'`) so it still is fingerprinting with CAN.

**Comment by sshane** [comment]
Now that we understand Hyundai platform codes, and can see that these cars accept different ECU queries, have different min steer speeds, merging the platforms in https://github.com/commaai/openpilot/pull/25951 wasn't correct. Reverting and keeping them separate platforms so we can add engine and not have to remove EPS for the GT/I30

**Comment by sshane** [comment]
Now that we understand Hyundai platform codes, and can see that these cars accept different ECU queries, have different min enable speeds, merging the platforms in https://github.com/commaai/openpilot/pull/25951 wasn't correct. Reverting and keeping them separate platforms so we can add engine and not have to remove EPS for the GT/I30

@sunnyhaibin can you have the user tell you what happens when they:

1) try to enable under 20 mph
3) let it come to a stop and it reaches 20 mph, does it disengage, keep braking? etc

**Pull request merged**